text/0178-mapping-to-otlp-anyvalue.md

```diff
@@ -0,0 +1,227 @@
+# Mapping Arbitrary Data to OTLP AnyValue
+
+This document defines how to map (convert) arbitrary data (e.g. in-memory
+objects) to OTLP's AnyValue.
+
+## Motivation
+
+The mapping is necessary to correctly implement Logging Library SDKs such that
+the converted values are unambiguous and consistent across languages and
+implementations.
+
+## Explanation
+
+[AnyValue](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L27)
+is capable of representing primitive and structured data of certain types.
+
+Implementations that have a source data in any form, such as in-memory objects
+or data coming from other formats that needs to be converted to AnyValue SHOULD
+follow the rules described below.
+
+### Primitive Values
+
+#### Integer Values
+
+Integer values which are within the range of 64 bit signed numbers
+[-2^63..2^63-1] SHOULD be converted to AnyValue's
+[int_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L33)
+field.
+
+Integer values which are outside the range of 64 bit signed numbers SHOULD be
+converted to AnyValue's
+[string_value](https://github.com/open-telemetry/opentelemetry-proto/blob/38b5b9b6e5257c6500a843f7fdacf89dd95833e8/opentelemetry/proto/common/v1/common.proto#L31)
```

Comment:
This mapping of many types into a single "string_value" type means this transformation is not going to be reversible.

Based on the PR description, when we say we want an unambiguous transformation we are only talking about one direction, right?

Reply:
Correct. Not reversible. Only "unambiguous" in the sense that it attempts to clearly say what to do. Maybe it is not the right word to use.